x-pack/plugins/observability/public/pages/slos/hooks/use_slo_summary.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import numeral from '@elastic/numeral';
+import { ALL_VALUE, SLOWithSummaryResponse } from '@kbn/slo-schema';
+import { paths } from '../../../../common/locators/paths';
+import { useKibana } from '../../../utils/kibana_react';
+import { NOT_AVAILABLE_LABEL } from '../../../../common/i18n';
+
+export const useSLOSummary = (slo: SLOWithSummaryResponse) => {
```

Comment:
This hook is doing a few different things. I don't mind extracting common reusable logic, but they need to be cohesive. 
I don't think the `sloDetailsUrl` and other values are very cohesive. Also I would rename this hook to `useSloFormattedSummary` or something similar.